### PR TITLE
required changes to include rmlui in bar widgets

### PR DESCRIPTION
**Pull request by loveridge**
### Work done
These are the minimum required changes in order to call RmlUi functions in bar widgets. These changes are required to allow rmlui widgets that can be used without a BAR development environment.

#### Test steps
Nothing should be changed when running BAR with just these changes. Adding an rmlui widget should show the new widgets ui.

### CODE REVIEW

**Comment by WatchTheFort** [comment]
What's the reason that all these font files need to be added?

**Comment by loveridge** [comment]
Support for bold, italic, and font weights.

**Comment by loveridge** [comment]
I went and found the bolditalic file

**Review by ChrisFloofyKitsune — APPROVED** [review]

**Comment by VLS-Volshok** [comment]
I think we need WTF for the review + merge, unless there's another core team member comfortable with it?

File: luaui/rml_setup.lua [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+-- Mouse Cursor Aliases
+--[[
+	These let standard CSS cursor names be used when doing styling.
+	If a cursor set via RCSS does not have an alias, it is unchanged.
+	CSS cursor list: https://developer.mozilla.org/en-US/docs/Web/CSS/cursor
+	RmlUi documentation: https://mikke89.github.io/RmlUiDoc/pages/rcss/user_interface.html#cursor
+]]
+
+-- when "cursor: normal" is set via RCSS, "cursornormal" will be sent to the engine... and so on for the rest
+RmlUi.SetMouseCursorAlias("default", 'cursornormal')
```
Comment (sprunk) [inline luaui/rml_setup.lua:64]: What about anything not listed here? E.g. "grab", what happens if I try to grab a panel to relocate it? Does the engine provide all known cursors by default?

File: luaui/rml_setup.lua [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+local function NewCreateContext(name)
+	local context = oldCreateContext(name)
+	context.dp_ratio = Spring.GetConfigFloat("ui_scale", 1)
+	return context
+end
+
+RmlUi.CreateContext = NewCreateContext
+
+-- Load fonts
+RmlUi.LoadFontFace("fonts/Poppins-Regular.otf", true)
```
Comment (sprunk) [inline luaui/rml_setup.lua:45]: What happens if this isn't done? Will the engine just fail to render or does it use default?

File: luaui/rml_setup.lua (hunk 1 above)
Comment (loveridge) [reply luaui/rml_setup.lua:64]: These are setting up the rml css `cursor` property to match normal css names (e.g. css `move` to spring `uimove`). You could alternatively just use the spring names in the css `cursor: uimove`. If they are not set, then the fallback for an invalid name is the default `cursornormal`

**Review by sprunk — APPROVED** [review]

**Comment by loveridge** [comment]
@sprunk Can I mark the conversations as resolved?

**Comment by Beherith** [comment]
Are there some RMLUI example widgets that could go in with this PR? Or was that added/proposed elsewhere?

**Comment by loveridge** [comment]
> Are there some RMLUI example widgets that could go in with this PR? Or was that added/proposed elsewhere?

These are the essentials to distribute rml widgets. Chris has a nice example widget that can have its own PR. Usually the comments are to split up changes further, and even this bare minimum change is taking weeks without that.

**Comment by Beherith** [comment]
Ive checked out chris's example branch, lovely stuff, managed to crash it in two minutes, and have now gone done the rabbit hole of updating stacktrace translator :D

**Pull request merged**